langchain4j-qianfan/src/main/java/dev/langchain4j/model/qianfan/InternalQianfanHelper.java

```diff
@@ -0,0 +1,221 @@
+                .parameters(toOpenAiParameters(toolSpecification.parameters()))
+                .build();
+    }
+
+    private static Parameters toOpenAiParameters(ToolParameters toolParameters) {
+        if (toolParameters == null) {
+            return Parameters.builder().build();
+        }
+        return Parameters.builder()
+                .properties(toolParameters.properties())
+                .required(toolParameters.required())
+                .build();
+    }
+    public static Message toErnieBotMessage(ChatMessage message) {
+
+        if (message instanceof UserMessage) {
+            UserMessage userMessage = (UserMessage) message;
+            return Message.builder()
+                    .role(Role.USER)
+                    .content(userMessage.text())
+                    .name(userMessage.name())
+                    .build();
+        }
+
+        if (message instanceof AiMessage) {
+            AiMessage aiMessage = (AiMessage) message;
+
+            if (!aiMessage.hasToolExecutionRequests()) {
+
+                return  Message.builder()
+                        .content(message.text())
+                        .role(Role.ASSISTANT)
+                        .build();
+            }
+
+            ToolExecutionRequest toolExecutionRequest = aiMessage.toolExecutionRequests().get(0);
+            if (toolExecutionRequest.id() == null) {
+                FunctionCall functionCall = FunctionCall.builder()
+                        .name(toolExecutionRequest.name())
+                        .arguments(toolExecutionRequest.arguments())
+                        .build();
+
+                return Message.builder()
+                        .content(message.text())
+                        .role(Role.ASSISTANT)
+                        .functionCall(functionCall)
+                        .build();
+            }
+
+        }
+        if (message instanceof ToolExecutionResultMessage) {
+            ToolExecutionResultMessage toolExecutionResultMessage = (ToolExecutionResultMessage) message;
+
+                FunctionCall functionCall = FunctionCall.builder()
+                        .name(toolExecutionResultMessage.toolName())
+                        .arguments(toolExecutionResultMessage.text())
+                        .build();
+                return  Message.builder()
+                        .content(message.text())
+                        .role(Role.FUNCTION)
+                        .name(functionCall.name())
+                        .build();
+
+        }
+        throw illegalArgument("Unknown message type: " + message.type());
+    }
+    static TokenUsage tokenUsageFrom(ChatCompletionResponse response) {
+        return Optional.of(response)
+                .map(ChatCompletionResponse::getUsage)
+                .map(usage -> new TokenUsage(usage.promptTokens(), usage.completionTokens(), usage.totalTokens()))
+                .orElse(new TokenUsage(null, null, null));
```

Comment:
If there is no token usage, please return `null` instead of empty `TokenUsage`. In other places as well